js/web/lib/wasm/session-options.ts

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import {InferenceSession} from 'onnxruntime-common';
+
+import {allocWasmString, iterateExtraOptions} from './options-utils';
+import {getInstance} from './wasm-factory';
+
+const getGraphOptimzationLevel = (graphOptimizationLevel: string|unknown): number => {
+  switch (graphOptimizationLevel) {
+    case 'disabled':
+      return 0;
+    case 'basic':
+      return 1;
+    case 'extended':
+      return 2;
+    case 'all':
+      return 99;
+    default:
+      throw new Error(`unsupported graph optimization level: ${graphOptimizationLevel}`);
+  }
+};
+
+const getExecutionMode = (executionMode: 'sequential'|'parallel'): number => {
+  switch (executionMode) {
+    case 'sequential':
+      return 0;
+    case 'parallel':
+      return 1;
+    default:
+      throw new Error(`unsupported execution mode: ${executionMode}`);
+  }
+};
+
+export const setSessionOptions = (options?: InferenceSession.SessionOptions): [number, number[]] => {
+  const wasm = getInstance();
+  let sessionOptionsHandle = 0;
+  const allocs: number[] = [];
+
+  const sessionOptions: InferenceSession.SessionOptions = options || {};
+
+  try {
+    if (options?.graphOptimizationLevel === undefined) {
+      sessionOptions.graphOptimizationLevel = 'all';
+    }
+    const graphOptimizationLevel = getGraphOptimzationLevel(sessionOptions.graphOptimizationLevel!);
+
+    if (options?.enableCpuMemArena === undefined) {
+      sessionOptions.enableCpuMemArena = true;
+    }
+
+    if (options?.enableMemPattern === undefined) {
+      sessionOptions.enableMemPattern = true;
+    }
+
+    if (options?.executionMode === undefined) {
+      sessionOptions.executionMode = 'sequential';
+    }
+    const executionMode = getExecutionMode(sessionOptions.executionMode!);
+
+    let logIdDataOffset = 0;
+    if (options?.logId !== undefined) {
+      logIdDataOffset = allocWasmString(options.logId, allocs);
+    }
+
+    if (options?.logSeverityLevel === undefined) {
+      sessionOptions.logSeverityLevel = 2;  // Default to warning
+    } else if (
+        typeof options.logSeverityLevel !== 'number' || options.logSeverityLevel < 0 || options.logSeverityLevel > 4) {
```

Comment:
Number.isInteger() would be better here. also applies to other integer config items